Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -295,13 +295,17 @@ WarpXParticleContainer::AddNParticles (int /*lev*/, long n,
         );
     }
 
+    // Move particles to their appropriate tiles
+    Redistribute();
+
     // Remove particles that are inside the embedded boundaries
 #ifdef AMREX_USE_EB
     auto & distance_to_eb = WarpX::GetInstance().GetDistanceToEB();
     scrapeParticles( *this, amrex::GetVecOfConstPtrs(distance_to_eb), ParticleBoundaryProcess::Absorb());
 #endif
 
-    Redistribute();
+    // Call (local) redistribute again to remove particles with invalid ids
+    Redistribute(lev, lev, 0, 1, true);
```

Comment:
Shouldn't this be moved inside the #ifdef:

```
#ifdef AMREX_USE_EB
    auto & distance_to_eb = WarpX::GetInstance().GetDistanceToEB();
    scrapeParticles( *this, amrex::GetVecOfConstPtrs(distance_to_eb), ParticleBoundaryProcess::Absorb());
    // Call (local) redistribute again to remove particles with invalid ids
    Redistribute(lev, lev, 0, 1, true);
#endif
```

Reply:
Yeah, that makes sense for sure. I'll update now.